content/posts/2025-02-26-whats_new_1_8.md

```diff
@@ -0,0 +1,295 @@
+          - action: Accept
+            cidr: 10.1.0.0/16
+            sampling: 1
+            pktDrops: true
+```
+
+<p style="text-align: center">Listing 6: Flow filter with packet drop example</p>
+
+
+### UDN Observability
+
+Kubernetes networking consists of a flat Layer 3 network and a single IP address space where every pod can communicate with any other pod.  In a number of use cases, this is undesirable.  OVN-Kubernetes provides another model called [User-Defined Networks](https://docs.openshift.com/container-platform/4.18/networking/multiple_networks/primary_networks/about-user-defined-networks.html) (UDN).  It supports microsegmentation where each network segment, which could be Layer 2 or Layer 3, is isolated from one another by the use of namespaces.
+
+UDN is a Developer Preview feature.  You must enable the feature gate named **NetworkSegmentation** (Listing 7).  Enter `oc edit featuregate` and update the **spec** section to:
+
+```
+spec:
+  featureSet: CustomNoUpgrade
+  customNoUpgrade:
+    enabled:
+      - NetworkSegmentation
+```
+
+<p style="text-align: center">Listing 7: Enable NetworkSegmentation feature gate</p>
+
+Similar to Network Events, changing the feature gate can take upwards of 10+ minutes for it to take effect.  You will lose connection to your cluster about halfway and when it's done, it will reconnect back online.  In OpenShift web console, it adds a new menu panel at **Networking > UserDefinedNetworks**.
+
+To enable this feature in FlowCollector, enter `oc edit flowcollector` and configure the following in the **ebpf** section:
+
+```
+spec:
+  agent:
+    ebpf:
+      sampling: 1  # recommended so all flows are observed
+      privileged: true
+      features:
+      - UDNMapping
+```
+
+<p style="text-align: center">Listing 8: Enable UDNMapping feature in FlowCollector</p>
+
+Now let's create a user-defined network.  You can use `oc apply` with the content in Listing 9, or copy and paste this into OpenShift web console.  To do the latter, click the **+** icon in the upper right corner next to the **?** icon, and select **Import YAML**.  Paste the YAML in and click **Create**.  You will now see the entry in **Networking > UserDefinedNetworks**.  Note: While you can create the UDN from this panel, it is very limiting in what you can configure.
+
+```
+apiVersion: k8s.ovn.org/v1
+kind: UserDefinedNetwork
+metadata:
+  name: udn-chiefs
+  namespace: chiefs
+spec:
+  topology: Layer2
+  layer2:
+    role: Primary
+    subnets:
+      - "10.0.0.0/24"
+```
+
+<p style="text-align: center">Listing 9: UDN - Layer 2 network segment</p>
+
+The next step is to add a secondary interface to a pod and have it join the UDN in the secondary network.  This is done using an annotation.  Set **k8s.v1.cni.cncf.io/networks** to the name of the UDN.  Assuming the pod is in a deployment, delete the pod and it will be recreated with two interfaces.  Check that there is a secondary interface.  See the commands in Listing 10.
+
```

Comment:
I'm not a UDN expert but I don't remember I had to do that to have pods in UDN. Just having the UDN defined, and restarting the pods was sufficient. Not sure if I am missing something ...

Reply:
Indeed that will not work as you need to annotate the namespace first using `k8s.ovn.org/primary-user-defined-network` and then create the UDN.

This will create the NAD and after all, you create the pods which automatically get attached to the network.

see https://docs.openshift.com/container-platform/4.18/networking/multiple_networks/primary_networks/about-user-defined-networks.html#nw-cudn-cr_about-user-defined-networks

This was a very recent change, just before the release...

Reply:
UDN in OCP 4.18 really changed, even with the RC version that I was using.  Therefore, I made a number of changes here.

- You no longer have to enable a feature gate as the feature is enabled.
- The label (not annotation) is on the namespace.  However, this is kind of bad because you can't add labels to an existing namespace.  You have to do it on creation of the namespace.